test/simple/test-regress-GH-14576.js

```diff
@@ -0,0 +1,47 @@
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/*
+ * Tests for sampler deadlock regression. 
+ * Issue https://github.com/joyent/node/issues/14576
+ */
+
+var assert = require('assert');
+var cp = require('child_process');
+
+var child = undefined;
+
+var deadlockTimer = setTimeout(function() {
+  if(child) {
+    process.kill(child.pid, 'SIGKILL');
+  }
+
+  assert(false, 'Node.js process hung');
+}, 10000);
+
+var testScript = "var i = 0; function r() { if(i++ > 2000) return; setTimeout(r, 1); }; r();";
```

Comment:
If I understand correctly, this is not sufficient as a regression test. The deadlock problem does not happen all the time and we would need to invoke node with `--prof` a certain number of times. I think 100 times is a safe assumption, as it's what I used to reproduce this issue and I was able to reproduce it within 100 runs all the time.

I also just validated that:
```
var i = 0; function r() { if(i++ > 2000) return; setTimeout(r, 1); }; r();
```
doesn't deadlock all the time with node v0.12.4:
```
➜  v0.12 git:(v0.12.4-release) ✗ ./node --version                                                                      
v0.12.4
➜  v0.12 git:(v0.12.4-release) ✗ time ./node -e 'var i = 0; function r() { if(i++ > 2000) return; setTimeout(r, 1); }; r();'
./node -e   0.12s user 0.07s system 7% cpu 2.580 total
➜  v0.12 git:(v0.12.4-release) ✗
```